kip-0010.md

```diff
@@ -0,0 +1,198 @@
+
+### 2. Shared Secret Scenario
+
+This scenario demonstrates a more complex script that allows for:
+
+1. Owner spending: The owner can spend the UTXO by providing a valid signature.
+2. Borrower spending with shared secret: A borrower can spend if they know a shared secret and create an output with a value greater than or equal to the input, sent to the same script.
+
+### 3. Mining Pool Payment Scenario
+
+This scenario demonstrates how the new opcodes can be used to optimize mining pool payouts, allowing for efficient M:N transactions where multiple inputs can be used to pay multiple participants.
+
+Concept:
+1. Pool Operation: The pool manages a KIP-10 compatible P2SH address for each participant, ensuring each address always has at least one UTXO.
+2. Payout Process: When a block is mined, the pool can efficiently distribute rewards using a single transaction with multiple inputs and outputs.
+
+The script for this scenario would allow two types of spending:
+1. Borrowing (used by the pool for payouts): Allows spending the UTXO if an output of greater value is created with the same script.
+2. Owner spending: Allows the participant to spend their funds normally with their signature.
+
+Advantages:
+1. Efficient M:N transactions: The pool can use M inputs (including the coinbase and participant UTXOs) to create N outputs (payouts to participants) in a single transaction.
+2. Reduced transaction costs: By combining multiple payouts into a single transaction, the overall transaction costs are significantly reduced.
+3. Improved payout frequency: The ability to efficiently batch payouts allows pools to distribute rewards more frequently without incurring excessive fees.
+4. Simplified pool operations: Pools can manage payouts more efficiently, streamlining their operational processes.
+5. Participant flexibility: Participants maintain full control over their funds while allowing the pool to optimize payouts.
+6. Enhanced scalability: As the number of pool participants grows, the efficiency gains from this approach become more pronounced.
+
+These examples demonstrate how the new opcodes enable complex spending conditions that were not possible before, allowing for innovative use cases such as threshold-based spending, shared control over UTXOs, and efficient mining pool operations.
+
+## Security Considerations
+
+1. Increased complexity in transaction validation, requiring careful implementation and testing.
+2. Potential for resource consumption attacks if scripts using these opcodes are not properly limited.
+3. Implications for transaction caching and optimizations, as scripts may now depend on broader transaction context.
+4. Potential privacy implications of allowing scripts to access more transaction data.
+
+Implementers should be aware of these considerations and implement appropriate safeguards, such as script size and complexity limits, to mitigate potential risks.
+
+## Related Work: Additive Addresses
```

Comment:
I think this can be a sub-section of the "Threshold Scenario". Also, a small explanation what "additive-only spending" means could be nice.